src/core/namespace.c

```diff
@@ -1303,6 +1401,32 @@ int setup_namespace(
                         return log_debug_errno(r, "Failed to decrypt dissected image: %m");
         }
 
+        if (n_mount_paths > 0) {
+                mount_path_loop_devices = new0(LoopDevice *, n_mount_paths);
+                if (!mount_path_loop_devices)
+                        return -ENOMEM;
+
+                for (i = 0; i < n_mount_paths; i++) {
+                        struct stat st;
+
+                        if (stat(mount_paths[i].source, &st) < 0)
+                                return -errno;
+
+                        if (S_ISBLK(st.st_mode)) /* Source is a block device */
+                                mount_path_loop_devices[i] = NULL;
```

Comment:
loop_device_make_by_path() has internal support for operating on existing block devices btw, you might as well just use that, and won#t need the stat() beforehand...